File: library/src/main/java/com/potyvideo/library/globalEnums/EnumPlayerSize.java
package com.potyvideo.library.globalEnums;

public enum EnumPlayerSize {

    UNDEFINE("UNDEFINE", -1),
    EXACTLY("EXACTLY", 0),
    AT_MOST("AT_MOST", 1),
    UNSPECIFIED("UNSPECIFIED", 2),
    ;

    private String valueStr;

    private Integer value;

    EnumPlayerSize(String valueStr, Integer value) {
        this.valueStr = valueStr;
        this.value = value;
    }

    public static EnumPlayerSize get(String value) {
        if (value == null) {
            return UNDEFINE;
        }

        EnumPlayerSize[] arr$ = values();
        for (EnumPlayerSize val : arr$) {
            if (val.valueStr.equalsIgnoreCase(value.trim())) {
                return val;
            }
        }

        return UNDEFINE;
    }

    public static EnumPlayerSize get(Integer value) {

        if (value == null) {
            return UNDEFINE;
        }

        EnumPlayerSize[] arr$ = values();
        for (EnumPlayerSize val : arr$) {
            if (val.value == value) {
                return val;
            }
        }

        return UNDEFINE;
    }

    public String getValueStr() {
        return valueStr;
    }

    public Integer getValue() {
        return value;
    }

    public void setValueStr(String valueStr) {
        this.valueStr = valueStr;
    }
}

File: library/src/main/java/com/potyvideo/library/globalEnums/EnumPlayerSize.kt
package com.potyvideo.library.globalEnums

enum class EnumPlayerSize(var valueStr: String, val value: Int) {

    UNDEFINE("UNDEFINE", -1),
    EXACTLY("EXACTLY", 0),
    AT_MOST("AT_MOST", 1),
    UNSPECIFIED("UNSPECIFIED", 2);

    companion object {

        operator fun get(value: String?): EnumPlayerSize {
            if (value == null) {
                return UNDEFINE
            }
            val `arr$` = values()
            for (`val` in `arr$`) {
                if (`val`.valueStr.equals(value.trim { it <= ' ' }, ignoreCase = true)) {
                    return `val`
                }
            }
            return UNDEFINE
        }

        operator fun get(value: Int?): EnumPlayerSize {
            if (value == null) {
                return UNDEFINE
            }
            val `arr$` = values()
            for (`val` in `arr$`) {
                if (`val`.value === value) {
                    return `val`
                }
            }
            return UNDEFINE
        }
    }
}